### PR TITLE
Feat/corl 2164 link to original commenter

### PR DESCRIPTION
**Pull request by marcushaddon**
## What does this PR do?
This PR will address https://vmproduct.atlassian.net/browse/CORL-2164 by causing the "In reply to..." text on replies to link the user back to the original comment. So far, this is successfully happening when the reply's parent is visible on the page (after Vinh's work on unread comments is merged, it will also cause the parent comment to be highlighted). 

If a comment's parent has not been loaded (ie it is "behind" a "Show more comments" button), the InReplyTo element should render as a non-actionable span element.

I do not believe there are any other edgecases that could cause a comment's parent to not be loaded, but if a user clicks this link and it is not found, it will be logged in Sentry.

Personal note: Currently the logic is "assume the comment's parent is loaded unless explicitly indicated otherwise". I might rename the property name to better reflect this.

## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## How do I test this PR?
Find a comment that is a reply to another comment and click "In Reply to...". The page should snap to the parent comment (currently only if its on the page).
 
 
## How do we deploy this PR?
No special actions should be required.

### CODE REVIEW

**Comment by alinhle** [comment]
I think we don't need to provide the jump to functionality inside of ConversationThread in the ancestor tree. The parent comment is right there (or sometimes after clicking on reveal) just above the comment.

**Comment by alinhle** [comment]
There is also some overlap here with the work I'm doing on the Comment Seen feature as you already pointed out.

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx
Comment (nick-funk) [inline src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx]: Might want to remove the eslint disables and console.log's

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx
Comment (nick-funk) [inline src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx]: More es-lint and console.log's that should probably be removed

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx
Comment (nick-funk) [inline src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx]: Thank you for using pym! This is one of the main things I was looking for! 💯

**Review by nick-funk — CHANGES_REQUESTED** [review]

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx
Comment (nick-funk) [reply src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx]: Make sure to run this border thing by design to check this is what they want. Super helpful for testing though!

File: src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx
Comment (cvle) [inline src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx:83]: I suggest renaming the property to _what it does_, e.g. `enableJumpToParent`

Comment (alinhle) [inline src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx]: I suggest renaming the property to _what it does_, e.g. `enableJumpToParent`. In your case because the `default` is `true`, I would actually call it `disableJumpToParent`.

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx
Comment (alinhle) [inline src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx]: Should use `useCallback` here and it also seems unnecessary to pass in the `id` because you already got this from `parent.id`.

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx
Comment (alinhle) [inline src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx]: I suggest here as well to rename the property to what it does like before: e.g. `enableJumpToParent` or `disableJumpToParent` depending on how you want the default to be.

File: src/core/client/stream/tabs/Comments/PermalinkView/ConversationThreadContainer.tsx
Comment (alinhle) [inline src/core/client/stream/tabs/Comments/PermalinkView/ConversationThreadContainer.tsx]: this shouldn't be `false`, but in this case with the `parentLoaded` property: `remaining === 0`

**Review by alinhle — CHANGES_REQUESTED** [review]

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx
Comment (nick-funk) [inline src/core/client/stream/tabs/Comments/Comment/InReplyTo.tsx:34]: ah, yes, that's what I forgot in my previous comment, to use pym to facilitate scrolling. Good job here!

**Review by nick-funk — APPROVED** [review]
Works well, I think this should be good to merge if we rebase after `commentSeen` stuff is merged into develop!

**Review by alinhle — APPROVED** [review]

File: src/core/client/stream/tabs/Comments/Comment/InReplyTo.css
Comment (alinhle) [inline src/core/client/stream/tabs/Comments/Comment/InReplyTo.css]: That's global css, which we shouldn't be using here.

**Review by alinhle — CHANGES_REQUESTED** [review]

**Review by alinhle — APPROVED** [review]

**Review by alinhle — APPROVED** [review]

**Pull request merged**
